src/main/java/com/hierynomus/smbj/smb2/messages/SMB2IoctlRequest.java

```diff
@@ -59,15 +57,24 @@ protected void writeTo(SMBBuffer smbBuffer) {
         fileId.write(smbBuffer);  // FileId (16 bytes)
 
         int offset = SMB2Header.STRUCTURE_SIZE + 56;
-        smbBuffer.putUInt32(offset); // InputOffset (4 bytes)
-        smbBuffer.putUInt32(inputData.length); // InputCount (4 bytes)
-        smbBuffer.putUInt32(MAX_OUTPUT_BUFFER_LENGTH); // MaxInputResponse (4 bytes)
+        if (inputData.length > 0) {
+            smbBuffer.putUInt32(offset); // InputOffset (4 bytes)
+            smbBuffer.putUInt32(inputData.length); // InputCount (4 bytes)
+        } else {
+            smbBuffer.putUInt32(0); // InputOffset (4 bytes)
```

Comment:
Must the offset be '0'? Or should it point to the location where the input would have been?